src/cpp/rtps/builtin/discovery/participant/PDPServer.cpp

```diff
@@ -0,0 +1,901 @@
+            rwatt.endpoint.topicKind = WITH_KEY;
+            rwatt.endpoint.durabilityKind = _durability; // Server Information must be persistent
+            rwatt.endpoint.reliabilityKind = RELIABLE;
+
+            // TODO: remove the join when Reader and Writer match functions are updated
+            rwatt.endpoint.remoteLocatorList.push_back(it->metatrafficMulticastLocatorList);
+            rwatt.endpoint.remoteLocatorList.push_back(it->metatrafficUnicastLocatorList);
+
+            mp_PDPReader->matched_writer_add(rwatt);
+        }
+
+    }
+    else
+    {
+        logError(RTPS_PDP, "PDPServer Reader creation failed");
+        delete(mp_PDPReaderHistory);
+        mp_PDPReaderHistory = nullptr;
+        delete(mp_listener);
+        mp_listener = nullptr;
+        return false;
+    }
+
+    hatt.payloadMaxSize = DISCOVERY_PARTICIPANT_DATA_MAX_SIZE;
+    hatt.initialReservedCaches = pdp_initial_reserved_caches;
+    hatt.memoryPolicy = mp_builtin->m_att.writerHistoryMemoryPolicy;
+    mp_PDPWriterHistory = new WriterHistory(hatt);
+
+    WriterAttributes watt;
+    watt.endpoint.endpointKind = WRITER;
+    watt.endpoint.durabilityKind = _durability;
+    watt.endpoint.properties.properties().push_back(Property("dds.persistence.plugin", "builtin.SQLITE3"));
+    watt.endpoint.properties.properties().push_back(Property("dds.persistence.sqlite3.filename",
+        GetPersistenceFileName()));
+    watt.endpoint.reliabilityKind = RELIABLE;
+    watt.endpoint.topicKind = WITH_KEY;
+    watt.endpoint.multicastLocatorList = mp_builtin->m_metatrafficMulticastLocatorList;
+    watt.endpoint.unicastLocatorList = mp_builtin->m_metatrafficUnicastLocatorList;
+    watt.times.heartbeatPeriod = pdp_heartbeat_period;
+    watt.times.nackResponseDelay = pdp_nack_response_delay;
+    watt.times.nackSupressionDuration = pdp_nack_supression_duration;
+
+    if (mp_RTPSParticipant->getRTPSParticipantAttributes().throughputController.bytesPerPeriod != UINT32_MAX &&
+        mp_RTPSParticipant->getRTPSParticipantAttributes().throughputController.periodMillisecs != 0)
+    {
+        watt.mode = ASYNCHRONOUS_WRITER;
+    }
+
+    if (mp_RTPSParticipant->createWriter(&mp_PDPWriter, watt, mp_PDPWriterHistory,
+        nullptr, c_EntityId_SPDPWriter, true))
+    {
+
+        for (auto it = mp_builtin->m_DiscoveryServers.begin(); it != mp_builtin->m_DiscoveryServers.end(); ++it)
+        {
+            RemoteReaderAttributes rratt;
+
+            rratt.guid = it->GetPDPReader();
+            rratt.endpoint.multicastLocatorList.push_back(it->metatrafficMulticastLocatorList);
+            rratt.endpoint.unicastLocatorList.push_back(it->metatrafficUnicastLocatorList);
+            rratt.endpoint.topicKind = WITH_KEY;
+            rratt.endpoint.durabilityKind = TRANSIENT_LOCAL;
+            rratt.endpoint.reliabilityKind = RELIABLE;
+
+            // TODO: remove the join when Reader and Writer match functions are updated
```

Comment:
because originally StatefulWriter::matched_reader_add used the endpoints remoteLocatorList. These lines are no longer needed and I proceed to remove them.